### PR TITLE
Translation haogao gu

### PR DESCRIPTION
**Pull request by Koohoko**
Translation from Haogao Gu (group 14).

contents:
_src_ (`zh_CN.po`)
|First line|	Last line|	
|2606|	2807|

_R_ (`R-zh_CN.po`)
|First line|	Last line|
|1264|	1415|

For the src (`zh_CN.po`), there are some chinese characteres (line 3056 - 3252) which have been previously added by other translator, appear to crash into nonsense characters on my machine, I am not sure if it is due to encoding problem of my editor. So I copy them from their previous commit on the web and paste back to my file. :(
Also I have difficult in triggering the message, so in fact I did not actually go through the testing. Sorry about that.

Thank you!

### CODE REVIEW

**Comment by MichaelChirico** [comment]
Excellent! Pending review from @biobai. I'll try and have a look into the encoding thing. Seems to be affecting everyone 😓

**Comment by MichaelChirico** [comment]
Have merged to the `translation` branch which covers the encoding thing. `update_pkg_po` working as intended -- all that's left is to confirm the Chinese itself! :)

**Comment by Koohoko** [comment]
@MichaelChirico Thank you!

File: po/zh_CN.po [hunk 1]
```diff
@@ -3092,48 +3092,48 @@ msgstr ""
...
 msgid ""
 "Internal error: Failed to allocate counts or TMP when assigning g in gforce"
-msgstr "�ڲ������� gforce ��Ϊ g ��ֵʱ��δ�ܳɹ�Ϊ counts ���� TMP ����ռ�"
+msgstr "内部错误：在 gforce 中为 g 赋值时，未能成功为 counts 或者 TMP 分配空间"
 
 #: gsumm.c:194
 #, c-format
 msgid "gforce assign high and low took %.3f\n"
-msgstr "gforce ���� high �� low ���� %.3f\n"
+msgstr "gforce 分配 high 和 low 用了 %.3f\n"
```
Comment (biobai) [inline po/zh_CN.po]: Glossary 文件里“assign”是翻译成“赋值”了

File: po/zh_CN.po [hunk 2]
```diff
@@ -3092,48 +3092,48 @@ msgstr ""
...
 #: gsumm.c:216
 msgid "gather took ... "
-msgstr "gather ���� ..."
+msgstr "gather 用了 ..."
 
 #: gsumm.c:334
 #, c-format
 msgid "gather implemented for INTSXP, REALSXP, and CPLXSXP but not '%s'"
-msgstr "gather ��֧�� INTSXP��REALSXP �� CPLXSXP������֧�� '%s'"
+msgstr "gather 已支持 INTSXP，REALSXP 和 CPLXSXP，但不支持 '%s'"
```
Comment (biobai) [inline po/zh_CN.po]: “implemented”此处是不是翻译“执行”？

File: po/zh_CN.po [hunk 3]
```diff
@@ -3142,133 +3142,133 @@ msgstr ""
...
 
 #: gsumm.c:565
 #, c-format
 msgid ""
 "Type '%s' not supported by GForce sum (gsum). Either add the prefix base::"
 "sum(.) or turn off GForce optimization using options(datatable.optimize=1)"
-msgstr "GForce ��ͣ�gsum����֧������ '%s'����ʹ�� base::sum(.) ����"
-"���� options(datatable.optimize=1) �ر� GForce �Ż�"
+msgstr "GForce 求和（gsum）不支持类型 '%s'。请使用 base::sum(.) 或者"
+"设置 options(datatable.optimize=1) 关闭 GForce 优化"
```
Comment (biobai) [inline po/zh_CN.po]: 建议“请使用前缀 base::sum(.) 或者设置 options(datatable.optimize=1) 来关闭 GForce 优化”

File: po/R-zh_CN.po [hunk 4]
```diff
@@ -1262,43 +1262,51 @@ msgid "length(na.last) > 1, only the first element will be used"
...
 msgid ""
 "input= must be a single character string containing a file name, a system "
 "command containing at least one space, a URL starting 'http[s]://', "
 "'ftp[s]://' or 'file://', or, the input data itself containing at least one "
 "\\n or \\r"
 msgstr ""
+"input= 必须是以下其中一种字符串: "
+"一个文件名, 一个含有不少于一个空格的系统命令, "
+"以'http[s]://','ftp[s]://' 或 'file://' 开头的URL, "
+"或是包含至少一个\\n 或 \\r的输入数据本身"
```
Comment (biobai) [inline po/R-zh_CN.po]: “或是包含至少一个\\n 或 \\r的输入数据本身” 建议为“或是本身就包含至少一个\\n 或 \\r的输入数据”

File: po/R-zh_CN.po [hunk 5]
```diff
@@ -1308,18 +1316,22 @@ msgid ""
 "the NEWS file for v1.11.6 for more information and for the option to "
 "suppress this message."
 msgstr ""
+"') 同时一个变量也被传输到 `input=`.请使用fread(cmd=...)."
+"这是出于安全考虑. 试想你在开发一个app, 这个app可能有恶意用户, "
+"同时这个app的运行环境不安全 (比如在root环境下运行)."
+"请阅读v1.11.6版本的NEWS file里面的item 5获取更多资讯, 或了解如何减少这个提示."
```
Comment (biobai) [inline po/R-zh_CN.po]: “或了解如何减少这个提示”建议为“或了解如何取消这个提示”

File: po/R-zh_CN.po [hunk 6]
```diff
@@ -1332,88 +1344,110 @@ msgid ""
...
 
 msgid ""
 "the metadata; please file an issue on GitHub if you'd like to see more "
 "intuitive behavior supported."
 msgstr ""
+"如果你希望更直观的功能, 请在 Github 上提交 issue."
 
 msgid "Encountered <"
-msgstr ""
+msgstr "遇到"
```
Comment (biobai) [inline po/R-zh_CN.po]: 这里漏掉了“<”

**Comment by biobai** [comment]
Hi @Koohoko , I finished the review process. Hope it helps.

**Comment by Koohoko** [comment]
> Hi @Koohoko , I finished the review process. Hope it helps.

Thanks so much for your kind help!

**Comment by Koohoko** [comment]
@MichaelChirico I agree with most of the comments, but what am I suppose to do to proceed? should I renew the code on my local machine and pull request again? Sorry I dont have experience like this before..

**Comment by MichaelChirico** [comment]
Yes, on your local copy, make the edits in the files. Then

```
# check the status of your files locally.
#   only po/R-zh_CN.po and po/zh_CN.po should appear
git status
# if that's working, commit these changes
git commit -am 'updated with feedback'
# push to GitHub; 'origin' is the name where your fork is.
#   You can confirm this by looking at git remote -v
#   origin should be in the line with Koohoko/data.table
git push origin translation-haogao-gu
```

**Comment by Koohoko** [comment]
@MichaelChirico My changes are in the "48bdf46" commit. However I seemed to messed up the changes when merging it with the previous edits (which lead to c8fda17).

Basically the "48bdf46" commit contains what I want, and I wish to abandon/revert the c8fda17 commit...

**Comment by MichaelChirico** [comment]
I _think_ what happened is you did a merge commit against the current `translation` branch and I _think_ I fixed it by:

 - reset your branch locally to the target commit: `git reset --hard 48bdf46639c204d73d9bb7896c017a7c546f6590`
 - force-push this back to remote: `git push haogao translation-haogao-gu --force`*
 - locally merge the `translation` branch into this branch (`translation-haogao-gu`): `git merge translation`
 - Resolve conflicts -- `po/zh_CN.po` and `po/R-zh_CN.po` were typical, but also two binary .mo files in `inst/po/zh_CN/LC_MESSAGES` were conflicted. I used `git checkout --ours -- inst/po/zh_CN/LC_MESSAGES/*` to "resolve" this conflict, but after finishing the merge, I re-ran `update_pkg_po('.')` and those `.mo` files changed again, so I suspect I should have done `--theirs` instead?

I'll merge now, but please have another glance and check everything looks OK.

* I think this removed some intermediate commits I had added to Haogao's _remote_ branch that hadn't been pulled locally? Or something? I was seeing a message in `git status` about being _5_ commits behind remote after doing `git reset --hard`, when I really should only have been _1_ commit behind. Not sure what those other commits were.

**Comment by Koohoko** [comment]
> I _think_ what happened is you did a merge commit against the current `translation` branch and I _think_ I fixed it by:
> 
> * reset your branch locally to the target commit: `git reset --hard 48bdf46639c204d73d9bb7896c017a7c546f6590`
> * force-push this back to remote: `git push haogao translation-haogao-gu --force`*
> * locally merge the `translation` branch into this branch (`translation-haogao-gu`): `git merge translation`
> * Resolve conflicts -- `po/zh_CN.po` and `po/R-zh_CN.po` were typical, but also two binary .mo files in `inst/po/zh_CN/LC_MESSAGES` were conflicted. I used `git checkout --ours -- inst/po/zh_CN/LC_MESSAGES/*` to "resolve" this conflict, but after finishing the merge, I re-ran `update_pkg_po('.')` and those `.mo` files changed again, so I suspect I should have done `--theirs` instead?
> 
> I'll merge now, but please have another glance and check everything looks OK.
> 
> * I think this removed some intermediate commits I had added to Haogao's _remote_ branch that hadn't been pulled locally? Or something? I was seeing a message in `git status` about being _5_ commits behind remote after doing `git reset --hard`, when I really should only have been _1_ commit behind. Not sure what those other commits were.

Thanks @MichaelChirico , I checked the translation in the most recent commit, they appear like what I want. 
I think I have to spend some time learning git, sometimes I am not clear what is happening...

**Pull request merged**